clap_complete_nushell/src/lib.rs

```diff
@@ -74,8 +74,22 @@ fn append_value_completion_and_help(
             Some(line) => indent.saturating_sub(line.len()),
             None => 0,
         };
-
-        s.push_str(format!("{:>width$}# {}", ' ', help).as_str());
+        let help = format!("{help}");
+        let mut lines = help.lines();
+        if let Some(first_line) = lines.next() {
+            // push the first line so we can know the indentation
+            s.push_str(format!("{:>width$}# {}", ' ', first_line).as_str());
+            if let Some(indent) = s
+                .lines()
+                .last()
+                .and_then(|last_line| last_line.chars().position(|c| c == '#'))
```

Comment:
Isn't the indentation just `width`?

Reply:
no, when I would use `width` it wouldn't align with the other comment, which nushell would probably parse fine (not sure on this), but it's a lot prettier i think
```patch
 export extern my-app [
     --test                    # This is a long description
-                              # which goes over 2 lines and ends with a dot.
+                  # which goes over 2 lines and ends with a dot.
 ]
```

Reply:
Hmm, overlooked that, thanks.

I think I'd prefer to adjust
```rust
        let width = match s.lines().last() {
            Some(line) => indent.saturating_sub(line.len()),
            None => 0,
        };
```
to
```rust
let (first_padding, rest_padding) = if let Some(line) s.lines.last() {
    let first = indent.saturating_sub(line.len());
    let rest = line.len() + first;
    (first, rest)
} else {
    (0, 0)
}
```
Then for the for loop, you can just do
```rust
for (i, line) in lines.iter().enumerate() {
    if i == 0 {
        let _ = write!("{:>first_padding$}# {line}", ' '));
    } else {
        let _ = write!("\n{:>rest_padding$}# {line}", ' '));
    }
}
```

I think that would make what the code is trying to do stand out a bit more.